File: src/data_structure/linked_list/merge_two.java
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
int FindMergeNode(Node headA, Node headB) {
    Node cur = headA;
    HashSet<Node> set = new HashSet<Node>();
    while (cur != null) {
        set.add(cur);
        cur = cur.next;
    }
    cur = headB;
    while (cur != null) {
        if (set.contains(cur)) {
            return cur.data;
        }
        cur = cur.next;
    }
    return 0;
}

File: src/data_structure/linked_list/merge_two.cpp
/*
   Find merge point of two linked lists
   Node is defined as
   struct Node
   {
       int data;
       Node* next;
   }
*/
int FindMergeNode(Node *headA, Node *headB)
{
    Node *target = headA, *source = headB;
    while (target != NULL) {
        while (source != NULL) {
            if (target->next == source->next && target->data == source->data) {
                return target->data;
            }
            source = source->next;
        }
        target = target->next;
        source = headB;
    }
    return 0;
}

